docs/painless/painless-guide/painless-execute-script.asciidoc

```diff
@@ -154,13 +255,566 @@ POST /_scripts/painless/_execute
+
+[source,console]
+----
+PUT /my-index-000001
+{
+  "mappings": {
+    "properties": {
+      "name": {
+        "type": "keyword"
+      },
+      "author": {
+        "type": "keyword"
+      },
+      "release_date": {
+        "type": "date"
+      },
+      "page_count": {
+        "type": "double"
+      }
+    }
+  }
+}
+----
+
+You can then write a script in the `boolean_field` context that indicates
+whether a book was published before the year 1972:
+
+[source,console]
+----
+POST /_scripts/painless/_execute
+{
+  "script": {
+    "source": """
+      emit(doc['release_date'].value.year < 1972);
+    """
+  },
+  "context": "boolean_field",
+  "context_setup": {
+    "index": "my-index-000001",
+    "document": {
+      "name": "Dune",
+      "author": "Frank Herbert",
+      "release_date": "1965-06-01",
+      "page_count": 604
+    }
+  }
+}
+----
+// TEST[continued]
+
+Because _Dune_ was published in 1965, the result returns as `true`:
+
+[source,console-result]
+----
+{
+  "result" : [
+    true
+  ]
+}
+----
+
+Similarly, you could write a script that determines whether the first name of
+an author exceeds a certain number of characters. The following script operates
+on the `author` field to determine whether the author's first name contains at
+least one character, but is less than five characters:
+
+[source,console]
+----
+POST /_scripts/painless/_execute
+{
+  "script": {
+    "source": """
+      int space = doc['author'].value.indexOf(' ');
+      emit(space > 0 && space < 5);
+    """
+  },
+  "context": "boolean_field",
+  "context_setup": {
+    "index": "my-index-000001",
+    "document": {
+      "name": "Dune",
+      "author": "Frank Herbert",
+      "release_date": "1965-06-01",
+      "page_count": 604
+    }
+  }
+}
+----
+// TEST[continued]
+
+Because `Frank` is five characters, the response returns `false` for the script
+valuation:
+
+[source,console-result]
+----
+{
+  "result" : [
+    false
+  ]
+}
+----
+
+[[painless-runtime-datetime]]
+===== `date_time`
+Several options are available for using
+<<painless-datetime,using datetime in Painless>>. In this example, you'll
+estimate when a particular author starting writing a book based on its release
+date and the writing speed of that author. The example makes some assumptions,
+but shows to write a script that operates on a date while incorporating
+additional information.
+
+Add the following fields to your index mapping to get started:
+
+[source,console]
+----
+PUT /my-index-000001
+{
+  "mappings": {
+    "properties": {
+      "name": {
+        "type": "keyword"
+      },
+      "author": {
+        "type": "keyword"
+      },
+      "release_date": {
+        "type": "date"
+      },
+      "page_count": {
+        "type": "long"
+      }
+    }
+  }
+}
+----
+
+The following script adds a variable named `pageTime` and makes an incredible
+assumption that this is an approximate time required to write a complete page
+for a book.
```

Comment:
What about something like:
The following script makes the incredible assumption that to write a book you just write each page and don't do research or revisions. Further, it assumes that the average time it takes to write a page is eight minutes.

But, yeah, I like the rewrite on this you did yesterday!